src/datagrid/DataGrid.tsx

```diff
@@ -243,6 +243,16 @@ export class DataGrid extends React.PureComponent<DataGridProps, DataGridState>
         if (this.props.pagination !== undefined) this.setInitalStateForPagination();
     }
 
+    componentWillReceiveProps(nextProps: DataGridProps) {
```

Comment:
componentWillReceiveProps is marked as depricated, So i think here we can use "getDerivedStateFromProps"instead of using componentWillReceiveProps.

Reply:
@vinodwani18  We know that and I have given a try for 'getDerivedStateFromProps' as well. But 'getDerivedStateFromProps' is a static method and we can not call instance methods from static. Even after making instance methods static we got some weird results for sorting and filtering. 
We have already discuss this internally and decided to go with 'componentWillReceiveProps' for now.
Along with 'componentWillReceiveProps' , 'componentWillMount' will also going to deprecate in future.  So we will log an enhancement bug for all common components to change deprecated methods.

Reply:
use componentDidUpdate

Reply:
@alexander-rassanov  'componentDidUpdate' life cycle hook gets execute after render() and we need some changes before that. Still I tried with it but it is no showing desired behavior. With this method the data grid is getting disappeared on any click.